File: 2022/16/java/Main.java
import java.io.IOException;

public class Main {
	// --- Constants ---
	private static final String INPUT_FILE = "../input/input.txt";

	public static void main(String[] args) throws IOException {
		Layout layout = Layout.load(INPUT_FILE);
		long total = layout.dfs1();
		System.out.format("1. Total pressure released: %,d%n", total);
		total = layout.dfs2();
		System.out.format("2. Total pressure released with an elephant: %,d%n", total);
	}
}

File: 2022/16/rust/src/main.rs
use rust::Layout;

const INPUT_FILE: &'static str = "../input/input.txt";

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let layout = Layout::load(INPUT_FILE)?;

    // --- Puzzle 1 ---
    let total = layout.dfs1();
    println!("1. Total pressure released: {}", total);

    // --- Puzzle 2 ---
    let total = layout.dfs2();
    println!("2. Total pressure released with an elephant: {}", total);

    Ok(())
}
